integration_tests/test_after.py

```diff
@@ -74,12 +75,10 @@ def test_build_with_missing_dependencies(self):
         with open('snapcraft.yaml', 'w') as snapcraft_yaml:
             snapcraft_yaml.write(wrong_contents)
 
-        exception = self.assertRaises(
-            subprocess.CalledProcessError,
+        self.assertRaises(
+            (errors.PluginCommandFailure, subprocess.CalledProcessError),
```

Comment:
This is an integration test, shelling out to Snapcraft. The only exception that it should be raising is `CalledProcessError`, no? And the returncode should be 2.

Reply:
No. One is raised by subprocess, the other by snapcraft. So we get two exceptions.

Reply:
Huh? We're shelling out, how can a Snapcraft process bubble past subprocess?

Reply:
subprocess raises CalledProcessError, which is caught. Afterwards Snapcraft raises the PluginCommandFailure (which is what this PR is all about)

Reply:
You're missing a layer. Integration tests call the snapcraft binary via `subprocess`. Then THAT snapcraft catches a PluginCommandFailure, causing it to exit with error code 2. That results in the integration test receiving a CalledProcessError from subprocess because the snapcraft it invoked exited non-zero.

Reply:
@kalikiana so, there are three exceptions at play here. But as @kyrofa said, `self.run_snapcraft` can only raise a subprocess exception. I think this test will just ignore the `PluginCommandFailure`.

This is the flow: `self.run_snapcraft` calls `snapcraft` through subprocess. `snapcraft` calls other tools throuhg `subprocess`, but that's something we don't control in this test. So, `snapcraft` raises a subprocess exception, that's transformed to a `snapcraft.errors` exception, that in the end is reported as exit code != 0. That causes `self.run_snapcraft` to raise a `subprocess` exception, and that last exception and exit code is the only thing we should check in integration tests.

Aside from that, there is a more fundamental problem in your test. Test should only check one thing, and you are checking two. If it were possible to check `errors.PluginCommandFailure` in this suite, that should be a separate test.

What you need is a unit test to check `PluginCommandFailure`.

Reply:
Great idea! I added separate unit tests.